kernel/src/main/scala/cats/kernel/instances/MapInstances.scala

```diff
@@ -47,11 +47,15 @@ import compat.scalaVersionSpecific._
 
 @suppressUnusedImportWarningForScalaVersionSpecific
 trait MapInstances extends MapInstances1 {
-  implicit def catsKernelStdHashForMap[K: Hash, V: Hash]: Hash[Map[K, V]] =
+  implicit def catsKernelStdHashForMapBinCompat[K, V: Hash]: Hash[Map[K, V]] =
```

Comment:
I'm actually not sure about this change. Even though the `Hash[K[` instance is not used, arguably it does serve as evidence that `K` _is_ hashable (e.g., it's not a `Function`).

See Oscar's interesting take on this in https://github.com/typelevel/cats/pull/4193#issuecomment-1119933018.

Reply:
Well, I'm ok to roll it back, but then another question is popping up actually: why the `MapHash` class uses the universal `.hashCode` on keys rather than `Hash[K]`? I am totally aware that Scala's `Map` itself uses universal hashes – but so what? The purpose of `MapHash` is to calculate `Hash[Map[K, V]]` out of the passed `Map` but it does not treat it as the "map", actually. All it does – just enumerates all its keys and values and produces a resulting hash value.

To me it seems that it would be better to make `MapHash` relying `Hash[K]` upon the calculation instead of the universal hash for `K`. Or are there come other considerations which I am just not aware of?

Reply:
> but so what?

😀 yeah, ok, great point. I think it dates back to when `Hash` was introduced in https://github.com/typelevel/cats/pull/1712 and I don't see any discussion on that specifically. Probably a separate PR, but I think it makes sense to change it (anyway we currently have a law claiming that `Hash` is consistent with `hashCode` so in theory it should not change anything).

Reply:
Ah, well, in theory you could end up with something inconsistent with `MapEq`? Since Scala forces our hand there to use universal equality.

https://github.com/typelevel/cats/blob/f8f6eff94872cba8bead3ae967d5b4f60f1b068d/kernel/src/main/scala/cats/kernel/instances/MapInstances.scala#L87-L97

But if we can't use the `Hash[K]`, I do see an argument we shouldn't ask for it. Ugh.

Reply:
So still maybe the best strategy for now is to keep the `Hash[K]` and suppress the warning. It still has some use, by proving `K` is hashable, and it leaves the door open for us to decide what we want to do.

Anyway, we have a real `HashMap` now so we shouldn't stress over this too much ... we should figure out how to land the other PR instead.